src/status_im2/common/floating_button_page/view.cljs

```diff
@@ -0,0 +1,113 @@
+(ns status-im2.common.floating-button-page.view
+  (:require
+    [oops.core :as oops]
+    [quo2.core :as quo]
+    [quo2.foundations.colors :as colors]
+    [react-native.blur :as blur]
+    [react-native.core :as rn]
+    [react-native.hooks :as hooks]
+    [react-native.platform :as platform]
+    [react-native.safe-area :as safe-area]
+    [reagent.core :as reagent]
+    [status-im2.common.floating-button-page.style :as style]))
+
+(def scroll-view-height (reagent/atom 0))
+(def content-container-height (reagent/atom 0))
+
+(defn show-button-background
+  [keyboard-height keyboard-shown content-scroll-y]
+  (let [button-container-height 64
+        keyboard-view-height    (+ keyboard-height button-container-height)]
+    (when keyboard-shown
+      (cond
+        platform/android?
+        (< (- @scroll-view-height button-container-height) @content-container-height)
+
+        platform/ios?
+        (< (- @scroll-view-height keyboard-view-height) (- @content-container-height content-scroll-y))
+
+        :else
+        false))))
+
+(defn button-container
+  [{:keys [show-keyboard? keyboard-shown show-background? keyboard-height]} children]
+  (let [insets (safe-area/get-insets)
+        height (reagent/atom 0)]
+    (reset! height (if show-keyboard? (if keyboard-shown keyboard-height 0) 0))
```

Comment:
This line looks weird

Reply:
True, most of this code is copied from the create profile page I'll clean it up a bit 🧽

Reply:
Yes, maybe:
```clojure
(reset! height (if (and show-keyboard? keyboard-shown)
                     keyboard-height
                     0))
```
but this is a reactive reset inside the render function, so I think it'd better to move it to another place.

Reply:
I mean what's the need for `show-keyboard?` when we have `keyboard-shown`